elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSearchResult.java

```diff
@@ -124,10 +124,22 @@ public String id() {
       return id;
     }
 
+    /**
+     * Find specific attribute from ES result or {@code null} if such
+     * attribute doesn't exists.
+     * @param name attribute name
+     * @return value from result or {@code null} if absent
+     */
+    Object valueOrNull(String name) {
+      Objects.requireNonNull(name, "name");
+      return sourceOrFields().containsKey(name) ? value(name) : null;
+    }
+
     /**
      * Finds specific attribute from ES search result
      * @param name attribute name
      * @return value from result (_source or fields)
+     * @throws IllegalArgumentException if such attribute doesn't exists
```

Comment:
This is being picky, but it should be "doesn't exist" (no s)

Reply:
This was addressed